spec/models/government_spec.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+RSpec.describe Government do
+  describe ".all" do
+    it "returns a collection of governments" do
+      expect(Government.all).to all be_a(Government)
+    end
+  end
+
+  describe ".find" do
+    let(:government) { build(:government, id: 1) }
+    before { allow(Government).to receive(:all).and_return([government]) }
+
+    it "finds a government by id" do
+      expect(Government.find(1)).to be government
+    end
+
+    it "returns nil if there isn't a government for the id" do
+      expect(Government.find(2)).to be_nil
+    end
+  end
+
+  describe ".for_date" do
+    let(:current_government) do
+      build(:government,
+            start_date: Date.parse("2018-12-01"),
+            end_date: nil)
+    end
+
+    let(:past_government) do
+      build(:government,
+            start_date: Date.parse("2018-01-31"),
+            end_date: Date.parse("2018-12-01"))
+    end
+
+    before do
+      allow(Government).to receive(:all)
+                       .and_return([current_government, past_government])
+    end
+
+    it "returns a government when there is one for the date" do
+      expect(Government.for_date(Date.parse("2019-06-01"))).to eq current_government
+    end
+
+    it "returns nil when there isn't a government for the date" do
+      expect(Government.for_date(Date.parse("2017-06-01"))).to be_nil
+    end
+
+    it "opts for the more recent government when there is a date conflict" do
+      expect(Government.for_date(Date.parse("2018-12-01"))).to eq current_government
+    end
+  end
+
+  describe ".current" do
+    let(:current_government) do
+      build(:government,
+            start_date: Date.parse("2018-12-01"),
+            end_date: nil)
+    end
+
+    let(:past_government) do
+      build(:government,
+            start_date: Date.parse("2018-01-31"),
+            end_date: Date.parse("2018-12-01"))
+    end
+
+    before do
+      allow(Government).to receive(:all)
+                       .and_return([current_government, past_government])
+    end
+
+    it "returns the current government" do
+      expect(Government.current).to eq current_government
+    end
+  end
+
+  describe "#covers?" do
+    let(:government) do
+      build(:government, start_date: start_date, end_date: end_date)
+    end
+
+    context "when there isn't an end date" do
+      let(:start_date) { Date.parse("2019-11-18") }
+      let(:end_date) { nil }
+
+      it "returns false before the start date" do
+        expect(government.covers?(Date.parse("2019-11-01"))).to be false
+      end
+
+      it "returns true on the start date" do
+        expect(government.covers?(Date.parse("2019-11-18"))).to be true
+      end
+
+      it "returns true after the start date" do
+        expect(government.covers?(Date.parse("2019-11-20"))).to be true
+      end
+    end
```

Comment:
I think the quirk you describe in your commit message (`A quirk I also found in the Whitehall data was that it wasn't fully...`) is being tested in the block above? I'd suggest encapsulating this code and explanation in its own commit. Keep the first commit simpler and then deal with the long tail of logic afterwards.